src/components/Hangar/Filter.js

```diff
@@ -76,24 +77,27 @@ export const Filter = (props) => {
   };
 
   const getStatsForCharts = (snapshots) => {
-    const statsForCharts = {};
-    snapshots.map((snapshot) => {
-      snapshot.lastBattleTime = new Date(snapshot.lastBattleTime * 1000).toLocaleDateString();
-      if (!statsForCharts[snapshot.lastBattleTime]) {
-        statsForCharts[snapshot.lastBattleTime] = { ...snapshot.regular };
+    const statsForCharts = snapshots.reduce((result, snapshot) => {
+      const lastBattleTime = new Date(snapshot.lastBattleTime * 1000).toLocaleDateString();
+      if (!result[lastBattleTime]) {
+        result[lastBattleTime] = { ...snapshot.regular };
       } else {
```

Comment:
Removed unnecessary check `if (!result[lastBattleTime])` and replaced it with the `result[lastBattleTime] = result[lastBattleTime] || { ...snapshot.regular };` statement.
```suggestion
        result[lastBattleTime] = result[lastBattleTime] || { ...snapshot.regular };
```